src/notation/view/notationviewinputcontroller.cpp

```diff
@@ -877,6 +881,10 @@ void NotationViewInputController::mouseReleaseEvent(QMouseEvent* event)
 
     if (interaction->isDragStarted()) {
         interaction->endDrag();
+        // When dragging of hairpin ends on a note or rest, open dynamic popup
```

Comment:
> on a note or rest

I don't see this reflected in the actual condition in the code. Is it a requirement at all?

Reply:
Actually the check for note or rest happens [here](https://github.com/musescore/MuseScore/pull/23038/files#diff-4dae0e75e0f35495b4862f2b2236a55da0caa3218daabe36a848349a635ea2f4R767-R770) in `Score::addText` which is called through `addTextToItem` in `toggleDynamicPopup`

Reply:
Ah okay. Still it might be better to adjust the comment, because it's a bit prone to causing confusion.